File: gluecodium/src/test/resources/smoke/comments/output/android/com/example/smoke/MapScene.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import com.example.NativeBase;
/**
 * <p>Referencing some type {@link com.example.smoke.MapScene#loadScene(int, MapScene.LoadSceneCallback)}.</p>
 */
public final class MapScene extends NativeBase {
    /**
     * @hidden
     */
    static class LoadSceneCallbackImpl extends NativeBase implements LoadSceneCallback {
        protected LoadSceneCallbackImpl(final long nativeHandle, final Object dummy) {
            super(nativeHandle, new Disposer() {
                @Override
                public void disposeNative(long handle) {
                    disposeNativeHandle(handle);
                }
            });
        }
        private static native void disposeNativeHandle(long nativeHandle);
        public native void apply(@Nullable final String p0);
    }
    @FunctionalInterface
    public interface LoadSceneCallback {
        void apply(@Nullable final String p0);
    }
    /**
     * For internal use only.
     * @hidden
     */
    protected MapScene(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    public native void loadScene(final int mapScheme, @Nullable final MapScene.LoadSceneCallback callback);
    public native void loadScene(@NonNull final String configurationFile, @Nullable final MapScene.LoadSceneCallback callback);
}

File: gluecodium/src/test/resources/smoke/comments/output/android-kotlin/com/example/smoke/MapScene.kt
/*

 *
 */

@file:JvmName("MapSceneExtensions")


package com.example.smoke

import com.example.NativeBase

/**
 * Referencing some type [com.example.smoke.MapScene.loadScene].
 */
class MapScene : NativeBase {

    fun interface LoadSceneCallback {


        fun apply(p0: String?) : Unit
    }

    /**
     * @suppress
     *
     * This class is used to represent C++ implementations of the interface or lambda in Kotlin.
     * It is instantiated by JNI and should not be used by the end users.
     */
    private class LoadSceneCallbackImpl : NativeBase, LoadSceneCallback {
        protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
            : super(nativeHandle, { disposeNativeHandle(it) }) {}



        override external fun apply(p0: String?) : Unit



        companion object {
            @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
        }
    }


    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}





    external fun loadScene(mapScheme: Int, callback: MapScene.LoadSceneCallback?) : Unit


    external fun loadScene(configurationFile: String, callback: MapScene.LoadSceneCallback?) : Unit




    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}


